plugins/out_stackdriver/stackdriver_conf.c

```diff
@@ -266,19 +336,50 @@ struct flb_stackdriver *flb_stackdriver_conf_create(struct flb_output_instance *
         ctx->metadata_server_auth = true;
     }
 
-    /* Resource type (only 'global' and 'gce_instance' are supported) */
-    tmp = flb_output_get_property("resource", ins);
+    /* set metadata server url with default, then replace if spec'd in config */
+    ctx->metadata_server = flb_sds_create(FLB_STD_META_URL);
+    tmp = flb_output_get_property("metadata_server", ins);
     if (tmp) {
-        if (validate_resource(tmp) != 0) {
-            flb_plg_error(ctx->ins, "unsupported resource type '%s'",
-                          tmp);
-            flb_stackdriver_conf_destroy(ctx);
-            return NULL;
+        ctx->metadata_server = flb_sds_create(tmp);
+    }
+
+    tmp = flb_output_get_property("resource", ins);
+    if (set_resource(tmp, ctx) != 0) {
+        flb_plg_error(ctx->ins, "unsupported resource type '%s'", tmp);
+        flb_stackdriver_conf_destroy(ctx);
+        return NULL;
+    }
+
+    /*
+     * Load monitored resource labels from config files.
+     *  Labels that don't match for a particular resource type will be
+     *  dropped since they will silently be rejected by the API anyway.
+     */
+    seen_project_id = false;
```

Comment:
Where is this actually set?

Reply:
I took it out for testing and never set it, but think I have a better way to do this now.